### PR TITLE
chore: add missing await to floating promises

### PR DESCRIPTION
**Pull request by yury-s**
We have quite a few places where returned promises have no catch block. Some of them are valid and never expected to throw (this are now marked with `void` operator) others were missing `await` or `catch` block. Fixing some of them to give an idea of what it would look like. If we fix remaining occurrences we can enable [`no-floating-promises`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-floating-promises.md) eslint check to prevent this problem in the future.

### CODE REVIEW

File: src/client/android.ts [hunk 1]
```diff
@@ -46,7 +46,7 @@ export class Android extends ChannelOwner<channels.AndroidChannel, channels.Andr
 
   setDefaultTimeout(timeout: number) {
     this._timeoutSettings.setDefaultTimeout(timeout);
-    this._channel.setDefaultTimeoutNoReply({ timeout });
+    void this._channel.setDefaultTimeoutNoReply({ timeout });
```
Comment (dgozman) [inline src/client/android.ts:49]: Let's do a catch for all NoReply methods?

File: src/server/chromium/crConnection.ts [hunk 2]
```diff
@@ -100,7 +100,7 @@ export class CRConnection extends EventEmitter {
     for (const session of this._sessions.values())
       session._onClosed(browserDisconnectedLogs);
     this._sessions.clear();
-    Promise.resolve().then(() => this.emit(ConnectionEvents.Disconnected));
+    void Promise.resolve().then(() => this.emit(ConnectionEvents.Disconnected));
```
Comment (dgozman) [inline src/server/chromium/crConnection.ts:103]: I don't quite like these voids for no real reason 😄

File: src/server/chromium/crNetworkManager.ts [hunk 3]
```diff
@@ -152,23 +152,23 @@ export class CRNetworkManager {
...
+      void this._client._sendMayFail('Fetch.continueRequest', {
         requestId: event.requestId
       });
     }
     if (!event.networkId) {
       // Fetch without networkId means that request was not recongnized by inspector, and
       // it will never receive Network.requestWillBeSent. Most likely, this is an internal request
       // that we can safely fail.
-      this._client._sendMayFail('Fetch.failRequest', {
+      void this._client._sendMayFail('Fetch.failRequest', {
```
Comment (dgozman) [inline src/server/chromium/crNetworkManager.ts:171]: Perhaps we should make `_sendMayFail` not return a promise?

File: src/server/chromium/crPage.ts [hunk 4]
```diff
@@ -646,7 +646,7 @@ class FrameSession {
     session.once('Runtime.executionContextCreated', async event => {
       worker._createExecutionContext(new CRExecutionContext(session, event.context));
     });
-    Promise.all([
+    void Promise.all([
```
Comment (dgozman) [inline src/server/chromium/crPage.ts:649]: Can just drop Promise.all.

File: src/server/chromium/crPage.ts [hunk 5]
```diff
@@ -969,7 +969,7 @@ class FrameSession {
     });
     const frameId = nodeInfo && typeof nodeInfo.node.frameId === 'string' ?
       nodeInfo.node.frameId : null;
-    documentElement.dispose();
+    void documentElement.dispose();
```
Comment (dgozman) [inline src/server/chromium/crPage.ts:972]: Same here, we can make handle.dispose() not return a Promise.

**Review by dgozman — COMMENTED** [review]
Overall, I think this adds more noise than solves problems. Some are good catches though. I'd advise against enabling the plugin.

File: src/server/chromium/crNetworkManager.ts (hunk 3 above)
Comment (yury-s) [reply src/server/chromium/crNetworkManager.ts:171]: There are methods like `InterceptableRequest.continue` that wait on the returned promise.

File: src/server/chromium/crPage.ts (hunk 4 above)
Comment (yury-s) [reply src/server/chromium/crPage.ts:649]: Yes, but it would be even more verbose as it'd require `void` prefix for each statement.

File: src/server/chromium/crPage.ts (hunk 5 above)
Comment (yury-s) [reply src/server/chromium/crPage.ts:972]: > Overall, I think this adds more noise than solves problems. Some are good catches though. I'd advise against enabling the plugin.

Yeah, I'm not fond of this `void` prefixes all over the code. Without them there is a lot of noise in the linter output masking real issues but we can live with that for now.

File: src/client/android.ts (hunk 1 above)
Comment (yury-s) [reply src/client/android.ts:49]: Done.

File: src/server/chromium/crConnection.ts (hunk 2 above)
Comment (yury-s) [reply src/server/chromium/crConnection.ts:103]: For these `void Promise.resolve().then(` calls we can either introduce a helper function that will do `.catch` internally or can add `.catch()` inline which is silly. Which way you like better?

File: src/server/chromium/crNetworkManager.ts (hunk 3 above)
Comment (yury-s) [reply src/server/chromium/crNetworkManager.ts:171]: Ok, updated the code.

File: src/server/chromium/crPage.ts (hunk 4 above)
Comment (yury-s) [reply src/server/chromium/crPage.ts:649]: Removed Promise.all and made replaced the methods with synchronous version.

File: src/server/chromium/crPage.ts (hunk 5 above)
Comment (yury-s) [reply src/server/chromium/crPage.ts:972]: Changed `dispose()` to not return promise.

File: src/client/android.ts (hunk 1 above)
Comment (yury-s) [reply src/client/android.ts:49]: Oops, only did it for the new ones, updating existing as well.

Comment (yury-s) [reply src/client/android.ts]: Since these are protocol calls, added .catch blocks.

File: src/server/chromium/crConnection.ts (hunk 2 above)
Comment (dgozman) [reply src/server/chromium/crConnection.ts:103]: I like neither 😄

File: src/server/firefox/ffExecutionContext.ts [hunk 6]
```diff
@@ -90,7 +90,7 @@ export class FFExecutionContext implements js.ExecutionContextDelegate {
     await this._session.send('Runtime.disposeObject', {
       executionContextId: this._executionContextId,
       objectId: handle._objectId,
-    }).catch(error => {});
+    });
```
Comment (dgozman) [inline src/server/firefox/ffExecutionContext.ts:93]: Why this change?

File: src/server/input.ts [hunk 7]
```diff
@@ -210,7 +210,7 @@ export class Mouse {
   async click(x: number, y: number, options: { delay?: number, button?: types.MouseButton, clickCount?: number } = {}) {
     const { delay = null, clickCount = 1 } = options;
     if (delay) {
-      this.move(x, y);
+      await this.move(x, y);
```
Comment (dgozman) [inline src/server/input.ts:213]: This was not awaiting on purpose.

File: src/server/snapshot/inMemorySnapshotter.ts [hunk 8]
```diff
@@ -53,11 +52,11 @@ export class InMemorySnapshotter extends BaseSnapshotStorage implements Snapshot
     await this._server.stop();
   }
 
-  async captureSnapshot(page: Page, snapshotName: string, element?: ElementHandle): Promise<SnapshotRenderer> {
+  async captureSnapshot(page: Page, snapshotName: string): Promise<SnapshotRenderer> {
     if (this._frameSnapshots.has(snapshotName))
       throw new Error('Duplicate snapshot name: ' + snapshotName);
 
-    this._snapshotter.captureSnapshot(page, snapshotName, element);
+    this._snapshotter.captureSnapshot(page, snapshotName);
```
Comment (dgozman) [inline src/server/snapshot/inMemorySnapshotter.ts:59]: Bad merge?

File: src/server/snapshot/snapshotter.ts [hunk 9]
```diff
@@ -183,26 +183,31 @@ export class Snapshotter {
...
-      (window as any)[kSnapshotStreamer].markIframe(frameElement, frameId);
-    }, { kSnapshotStreamer, frameElement, frameId: frame.uniqueId });
-    frameElement.dispose();
-  } catch (e) {
-  }
+function annotateFrameHierarchy(frame: Frame) {
+  _annotateFrameHierarchyAsync(frame).catch(e => {});
+}
+
+async function _annotateFrameHierarchyAsync(frame: Frame) {
```
Comment (dgozman) [inline src/server/snapshot/snapshotter.ts:201]: Reminds me of C# 😄

File: src/server/supplements/inspectorController.ts [hunk 10]
```diff
@@ -45,15 +45,15 @@ export class InspectorController implements InstrumentationListener {
         case 'log':
           const originalMetadata = this._waitOperations.get(info.waitId)!;
           originalMetadata.log.push(info.message);
-          this.onCallLog('api', info.message, sdkObject, originalMetadata);
+          await this.onCallLog('api', info.message, sdkObject, originalMetadata);
```
Comment (dgozman) [inline src/server/supplements/inspectorController.ts:48]: I think these were without await on purpose.

File: src/server/supplements/recorderSupplement.ts [hunk 11]
```diff
@@ -402,18 +402,18 @@ export class RecorderSupplement {
...
-      this._snapshotter.captureSnapshot(sdkObject.attribution.page, snapshotName);
+      await this._snapshotter.captureSnapshot(sdkObject.attribution.page, snapshotName);
     }
   }
 
   async onBeforeCall(sdkObject: SdkObject, metadata: CallMetadata): Promise<void> {
     if (this._mode === 'recording')
       return;
-    this._captureSnapshot(sdkObject, metadata, 'before');
+    await this._captureSnapshot(sdkObject, metadata, 'before');
```
Comment (dgozman) [inline src/server/supplements/recorderSupplement.ts:416]: It is important to no await here.

File: src/server/webkit/wkPage.ts [hunk 12]
```diff
@@ -113,7 +113,7 @@ export class WKPage implements PageDelegate {
     promises.push(this.updateHttpCredentials());
     if (this._browserContext._permissions.size) {
       for (const [key, value] of this._browserContext._permissions)
-        this._grantPermissions(key, value);
+        promises.push(this._grantPermissions(key, value));
```
Comment (dgozman) [inline src/server/webkit/wkPage.ts:116]: Good catch!

File: src/server/input.ts (hunk 7 above)
Comment (yury-s) [reply src/server/input.ts:213]: Sounds sketchy

File: src/server/snapshot/inMemorySnapshotter.ts (hunk 8 above)
Comment (yury-s) [reply src/server/snapshot/inMemorySnapshotter.ts:59]: No, that parameter is unused.

File: src/server/supplements/inspectorController.ts (hunk 10 above)
Comment (yury-s) [reply src/server/supplements/inspectorController.ts:48]: should be .catch then?

File: src/server/supplements/recorderSupplement.ts (hunk 11 above)
Comment (yury-s) [reply src/server/supplements/recorderSupplement.ts:416]: hm, this should be commented then, maybe .catch too?

File: src/server/firefox/ffExecutionContext.ts (hunk 6 above)
Comment (yury-s) [reply src/server/firefox/ffExecutionContext.ts:93]: It's caller's responsibility to catch exceptions if any and it does so now in .dispose().

**Comment by yury-s** [comment]
Removed controversial parts, left only simple ones.

**Review by dgozman — APPROVED** [review]
Looks great!
